### PR TITLE
Dependency update

### PR DESCRIPTION
**Pull request by chapj2001**
Update dependencies and shade cassandra-all usage to ease migration from Astyanax to DataStax Java Driver for Apache Cassandra. This PR does not signal any change in Astyanax deprecation.

Astyanax is deprecated but still used by many projects. By now Astyanax's very old dependencies often increase the effort required to migrate to the DataStax driver. The transition can be especially difficult for projects that need both Astyanax and DataStax drivers to coexist for some time, e.g. an app that wants to migrate its own Astyanax dependency to DataStax driver but still has other transitive dependencies to Astyanax outside its direct control.

Summary of changes and impact to Astyanax users:
 * Shaded cassandra-all to avoid dependency incompatibilities with newer versions of cassandra-unit/CassandraDaemon/EmbeddedCassandra for unit/integration testing. Relocates astyanax-cassandra's "org.apache.cassandra" dependencies to "com.netflix.astyanax.org.apache.cassandra."
 * Updated gradle wrapper to 4.1
 * Updated all explicit dependencies to newer compatible versions.
 * Unit-tested with Guava versions 19-25.
 * Eliminated/refactored some dependency declarations to reflect true dependencies and minimize transitive dependencies forced on Astyanax users.
 * Moved dependencies not actually used by src/main to "testCompile" to keep them out of transitive dependencies.
 * Updated readme to more explicitly suggest Astyanax users migrate to DataStax driver.
 * Note that with astyanax-shaded-cassandra-all, currently IntelliJ incorrectly renders "com.netflix.astyanax.org.apache.cassandra." as unfound. Astyanax still builds perfectly fine via Gradle. This is a known IntelliJ bug which affects **only** the Astyanax modules themselves; users of Astyanax artifacts will not see any issue.
 * Added missing license headers.

### CODE REVIEW

File: astyanax-cassandra/src/main/java/com/netflix/astyanax/serializers/ComparatorType.java [hunk 1]
```diff
@@ -15,45 +15,43 @@
...
-import com.netflix.astyanax.*;
+import com.netflix.astyanax.Serializer;
+import com.netflix.astyanax.shaded.org.apache.cassandra.db.marshal.ShadedTypeParser;
+
+import static com.netflix.astyanax.shaded.org.apache.cassandra.db.marshal.ShadedTypeParser.SHADED_PREFIX;
 
 /**
  * @author: peter
  */
 public enum ComparatorType {
```
Comment (vinaykumarchella) [inline astyanax-cassandra/src/main/java/com/netflix/astyanax/serializers/ComparatorType.java]: Would this comparator come from C*? or client side?

File: astyanax-cql/src/main/java/com/netflix/astyanax/cql/retrypolicies/ChangeConsistencyLevelRetryPolicy.java [hunk 2]
```diff
@@ -112,6 +114,22 @@ public RetryDecision onUnavailable(Statement query, ConsistencyLevel cl,
 			boolean shouldRetry = retryOnAllConditions || retryOnUnavailable;
 			return checkRetry(query, cl, shouldRetry);
 		}
+
```
Comment (vinaykumarchella) [inline astyanax-cql/src/main/java/com/netflix/astyanax/cql/retrypolicies/ChangeConsistencyLevelRetryPolicy.java]: Is this change needed as part of this PR?

File: astyanax-cassandra/src/main/java/com/netflix/astyanax/serializers/ComparatorType.java (hunk 1 above)
Comment (chapj2001) [reply astyanax-cassandra/src/main/java/com/netflix/astyanax/serializers/ComparatorType.java]: Doesn't really matter. This comparator is just responsible for returning the correct serializer regardless of whether a given type's class is in the original Cassandra package (e.g. handling messages to/from C* cluster) or the shaded Cassandra package (e.g. code within Astyanax). It's the core guts of what makes Astayanax's shaded cassandra-all talk to other processes which use unshaded original cassandra-all.

File: astyanax-cql/src/main/java/com/netflix/astyanax/cql/retrypolicies/ChangeConsistencyLevelRetryPolicy.java (hunk 2 above)
Comment (chapj2001) [reply astyanax-cql/src/main/java/com/netflix/astyanax/cql/retrypolicies/ChangeConsistencyLevelRetryPolicy.java]: Yes, it provides backwards-compatibility for the previous version's cql retry policy by handling the new mandatory onRequestError() interface introduced by the upgrade to cassandra-driver-core 3.3.2.

**Comment by chapj2001** [comment]
re: the suggestion for more unit tests for all the datatype changes, the vast majority of those changes are very simple (reference shaded package of a class from cassandra-all) and they're pretty deep in astyanax-core so that they already git hit pretty well by existing unit tests from astyanax-test. I'll see if I can find some code coverage metric to verify that we covered each datatype at least once.
